Source/MLX/DType.swift

```diff
@@ -5,7 +5,25 @@ import Numerics
 
 /// Enum wrapping `Cmlx.mlx_array_dtype`.
 ///
-/// This is typically not used directly, rather it is inferred from parameters that are ``HasDType``.
+/// This is typically not used directly, rather it is inferred from parameters that are ``HasDType``, for example:
+///
+/// ```swift
+/// let ones = MLXArray.ones([2, 2], type: Int32.self)
+/// ```
+///
+/// provides ``int32`` via ``HasDType/dtype``.
+///
+/// Some methods take a `dtype` directly for cases where it may not be possible to use a Swift type:
+///
+/// ```swift
+/// let bf = MLXArray(10.5, dtype: .bfloat16)
```

Comment:
Somewhat unrelated to this change, but it's a bit of a mystery to me why one of these functions takes a `type` and the other takes a `dtype`. Are they different types?

Reply:
Yeah, `type:` takes a swift type, e.g. `.item(Int.self)`.  This is more typical Swift API (passing types rather than enums) but more importantly it allows me to write a function like this:

```
def item(T: HasDType)(type: T.Type) -> T
```

By using the swift type as a parameter I can make a function that returns that type (generically).